hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsV_0_34Module.java

```diff
@@ -178,44 +182,52 @@ static Operation bindDelegateCallOperation(
     static Operation bindStaticCallOperation(
             final GasCalculator gasCalculator,
             @V_0_34 final BiPredicate<Address, MessageFrame> addressValidator,
-            final Map<String, PrecompiledContract> precompiledContractMap) {
-        return new HederaStaticCallOperation(gasCalculator, addressValidator, precompiledContractMap);
+            final @Named("HederaSystemAccountDetector") Predicate<Address> hederaSystemAccountDetector) {
+        return new HederaStaticCallOperation(gasCalculator, addressValidator, hederaSystemAccountDetector);
     }
 
     @Provides
     @Singleton
     @IntoSet
     @ContractsModule.V_0_34
     static Operation bindBalanceOperation(
-            GasCalculator gasCalculator, @ContractsModule.V_0_34 BiPredicate<Address, MessageFrame> addressValidator) {
-        return new HederaBalanceOperation(gasCalculator, addressValidator);
+            GasCalculator gasCalculator,
+            @ContractsModule.V_0_34 BiPredicate<Address, MessageFrame> addressValidator,
+            final @Named("HederaSystemAccountDetector") Predicate<Address> hederaSystemAccountDetector) {
```

Comment:
similar to v30, we should not be injecting a system contract detector.  it's always false.  For compatibility of archive node evaluations.